File: Examples/VisualizationAlgorithms/Python/VisQuad.py
#!/usr/bin/env python

# This example demonstrates the use of the contour filter, and the use of
# the vtkSampleFunction to generate a volume of data samples from an
# implicit function.

import vtk

# VTK supports implicit functions of the form f(x,y,z)=constant. These 
# functions can represent things spheres, cones, etc. Here we use a 
# general form for a quadric to create an elliptical data field.
quadric = vtk.vtkQuadric()
quadric.SetCoefficients(.5, 1, .2, 0, .1, 0, 0, .2, 0, 0)

# vtkSampleFunction samples an implicit function over the x-y-z range
# specified (here it defaults to -1,1 in the x,y,z directions).
sample = vtk.vtkSampleFunction()
sample.SetSampleDimensions(30, 30, 30)
sample.SetImplicitFunction(quadric)

# Create five surfaces F(x,y,z) = constant between range specified. The
# GenerateValues() method creates n isocontour values between the range
# specified.
contours = vtk.vtkContourFilter()
contours.SetInput(sample.GetOutput())
contours.GenerateValues(5, 0.0, 1.2)

contMapper = vtk.vtkPolyDataMapper()
contMapper.SetInput(contours.GetOutput())
contMapper.SetScalarRange(0.0, 1.2)

contActor = vtk.vtkActor()
contActor.SetMapper(contMapper)

# We'll put a simple outline around the data.
outline = vtk.vtkOutlineFilter()
outline.SetInput(sample.GetOutput())

outlineMapper = vtk.vtkPolyDataMapper()
outlineMapper.SetInput(outline.GetOutput())

outlineActor = vtk.vtkActor()
outlineActor.SetMapper(outlineMapper)
outlineActor.GetProperty().SetColor(0,0,0)

# The usual rendering stuff.
ren = vtk.vtkRenderer()
renWin = vtk.vtkRenderWindow()
renWin.AddRenderer(ren)
iren = vtk.vtkRenderWindowInteractor()
iren.SetRenderWindow(renWin)

ren.SetBackground(1, 1, 1)
ren.AddActor(contActor)
ren.AddActor(outlineActor)

iren.Initialize()
renWin.Render()
iren.Start()

File: graphics/examplesCxx/VisQuad.cxx
#include "vtk.h"

main ()
{
  vtkRenderMaster rm;
  vtkRenderWindow *renWin;
  vtkRenderer *ren1;
  vtkRenderWindowInteractor *iren;
  vtkActor *outlineActor;
  vtkActor *contActor;
  vtkPolyMapper *outlineMapper, *contMapper;
  vtkSampleFunction *sample;
  vtkQuadric *quadric;
  vtkOutlineFilter *outline;
  vtkContourFilter *contours;

  renWin = rm.MakeRenderWindow();
  iren = renWin->MakeRenderWindowInteractor();
  ren1 = renWin->MakeRenderer();

  // Quadric definition
  quadric = new vtkQuadric;
    quadric->SetCoefficients(.5,1,.2,0,.1,0,0,.2,0,0);

  sample = new vtkSampleFunction;
    sample->SetSampleDimensions(50,50,50);
    sample->SetImplicitFunction(quadric);

  // Create five surfaces F(x,y,z) = constant between range specified
  contours = new vtkContourFilter;
    contours->SetInput(sample->GetOutput());
    contours->GenerateValues(5, 0.0, 1.2);

  contMapper = vtkPolyMapper::New();
    contMapper->SetInput(contours->GetOutput());
    contMapper->SetScalarRange(0.0, 1.2);

  contActor = vtkActor::New();
    contActor->SetMapper(contMapper);

  // Create outline
  outline = new vtkOutlineFilter;
    outline->SetInput(sample->GetOutput());

  outlineMapper = vtkPolyMapper::New();
    outlineMapper->SetInput(outline->GetOutput());

  outlineActor = vtkActor::New();
    outlineActor->SetMapper(outlineMapper);
    outlineActor->GetProperty()->SetColor(0,0,0);

  ren1->SetBackground(1,1,1);
  ren1->AddActors(contActor);
  ren1->AddActors(outlineActor);

  renWin->Render();

  // interact with data
  iren->Start();
}
